libqtile/widget/windowtabs.py

```diff
@@ -68,22 +63,21 @@ def _configure(self, qtile, bar):
     def update(self, *args):
         names = []
         for w in self.bar.screen.group.windows:
+            try:
+                name = self.parse_text(w.name if w and w.name else "")
+            except:  # noqa: E722
+                logger.exception("parse_text function failed:")
+                name = w.name if w and w.name else "(unnamed)"
```

Comment:
Do you need the `if w` part of the check here? If there any scenario where `w` is not a window object?

Reply:
That I'm not sure. It comes from the original implementation, so I kept it.